README.md

```diff
@@ -26,6 +26,100 @@ Manticore can analyze the following types of programs:
 Manticore is supported on Linux and requires Python >=3.6. Ubuntu 18.04 is strongly recommended.
 Ethereum smart contract analysis requires the [`solc`](https://github.com/ethereum/solidity) program in your `$PATH`.
 
+## Usage
+
+### CLI
+
+Manticore has a command line interface which can be used to easily symbolically execute a supported program. Analysis results will be placed into a new directory beginning with `mcore_`.
+
+
+```
+$ manticore ./path/to/binary        # runs, and creates a mcore_* directory with analysis results
+$ manticore ./path/to/binary ab cd  # use concrete strings "ab", "cd" as program arguments
+$ manticore ./path/to/binary ++ ++  # use two symbolic strings of length two as program arguments
+```
+
+### API
+
+Manticore has a Python programming interface which can be used to implement custom analyses.
+
+
+```python
+# example Manticore script
+from manticore import Manticore
+
+hook_pc = 0x400ca0
+
+m = Manticore('./path/to/binary')
+
+@m.hook(hook_pc)
+def hook(state):
+  cpu = state.cpu
+  print('eax', cpu.EAX)
+  print(cpu.read_int(cpu.ESP))
+
+  m.terminate()  # tell Manticore to stop
+
+m.run()
+```
+
+### Ethereum
+Manticore ships with a library of vulnerability finding analyses accessible from the command line. Solidity smart contracts must have a `.sol` extension to be consumed by Manticore.
+
+```
+$ manticore ./path/to/contract.sol  # runs, and creates a mcore_* directory with analysis results
+$ manticore --detect-reentrancy ./path/to/contract.sol  # Above, but with reentrancy detection enabled
+$ manticore --detect-all ./path/to/contract.sol  # Above, but with all bug detectors enabled
+```
+
+Manticore also contains a python API specifically designed for analyzing smart contracts.
+
+```python
+from manticore.ethereum import ManticoreEVM
+contract_src="""
+contract Contract {
+    function foo(uint value) public returns (uint){
+        if(value==1) revert();
+        return value+1;
+    }
+}
+"""
+m = ManticoreEVM()
+
+user_account=m.create_account(balance=1000)
+contract_account=m.solidity_create_contract(contract_src,
+                                            owner=user_account,
+                                            balance=0)
+bar=m.make_symbolic_value(name="bar")
```

Comment:
You do not actually need to provide a `name=` there (in case you find it more clear not to)